src/core/contracts/identity.test.ts

```diff
@@ -331,74 +339,358 @@ describe("identity", () => {
-        expect(await getRemoveDelegateLogData(fakeAddress)).toEqual([]);
+        expect(result).toEqual([]);
       });
     });
 
-    describe("when logs have been emmited for delegate", () => {
+    describe("when a public address is associated (belongs) to a single identity contract", () => {
+      let contractOwner: EthereumAddress;
+      let contractAddress: EthereumAddress;
+      let signer: ethers.Signer;
+
       beforeAll(async () => {
+        await snapshotHardhat(provider);
+      });
+
+      beforeAll(async () => {
+        signer = getSignerForAccount(10);
         contractOwner = await signer.getAddress();
+
         const identityContract = await new Identity__factory(signer).deploy(contractOwner);
         await identityContract.deployed();
         contractAddress = identityContract.address;
+      });
+
+      afterAll(async () => {
+        await revertHardhat(provider);
+      });
+
+      it("returns one identity contract address associated with public address", async () => {
+        const result = await getDelegeteIdentitiesFor(contractOwner);
+
+        expect(result.length).toBe(1);
+        expect(result).toEqual([contractAddress]);
+      });
+
+      describe("and an update is made to change permission", () => {
+        beforeEach(async () => {
+          await upsertDelegate(contractAddress, contractOwner, 0x1, { signer: signer });
+        });
+
+        it("returns one identity contract address associated with public address", async () => {
+          const result = await getDelegeteIdentitiesFor(contractOwner);
+
+          expect(result.length).toBe(1);
+          expect(result).toEqual([contractAddress]);
+        });
+      });
+
+      describe("and delegate is removed immediately", () => {
+        beforeAll(async () => {
+          await snapshotHardhat(provider);
+        });
+
+        afterAll(async () => {
+          await revertHardhat(provider);
+        });
+
+        beforeAll(async () => {
+          await removeDelegate(contractAddress, contractOwner, 0x1, { signer: signer });
+        });
+
+        it("returns zero identity contract address associated with public address", async () => {
+          const result = await getDelegeteIdentitiesFor(contractOwner);
+
+          expect(result.length).toBe(0);
+          expect(result).toEqual([]);
+        });
+      });
+
+      describe("and delegate is scheduled to be removed after 10 blocks", () => {
+        beforeAll(async () => {
+          await snapshotHardhat(provider);
+        });
+
+        afterAll(async () => {
+          await revertHardhat(provider);
+        });
+
+        beforeEach(async () => {
+          const curentBlockNumber = provider.blockNumber;
+          await removeDelegate(contractAddress, contractOwner, curentBlockNumber + 0x10, { signer: signer });
+        });
+      });
+
+      it("continues to returns identity contract address associated with public address", async () => {
```

Comment:
Is there a way to make a test that advances the blockNumber and show it removes the delegate?

Reply:
Yeah. I originally added code to fast forward the the blocks and then reset it but then I thought to myself perhaps would not be necessary since I have this test: https://github.com/LibertyDSNP/sdk-ts/pull/108/files#diff-d27b34aec361b803a3d5945f01cfb662a386cfaccecd6f6e6866be75baf76fa7R395

Reply:
let me know if you want me to add it back on.